patches/server/0263-Improve-death-events.patch

```diff
@@ -263,10 +259,66 @@ index 224eca7d20cf4b890a6bc1b314d566e02e716762..7281eb294ddd178ba742088d3c61bf3d
      public void addAdditionalSaveData(CompoundTag nbt) {
          super.addAdditionalSaveData(nbt);
 diff --git a/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java b/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java
-index d545349f659b2a164a28d06e9ff0f9fff8fa8ecf..bbde9b758643c087733064a126d90689d71830cf 100644
+index d545349f659b2a164a28d06e9ff0f9fff8fa8ecf..011c54d32122b5c125c8c30224bfa4b3c041fbf7 100644
 --- a/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java
 +++ b/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java
-@@ -755,7 +755,8 @@ public class ArmorStand extends LivingEntity {
+@@ -499,7 +499,7 @@ public class ArmorStand extends LivingEntity {
+                 // CraftBukkit end
+                 if (source.isExplosion()) {
+                     this.brokenByAnything(source);
+-                    this.kill();
++                    // this.kill(); // Paper - Prevent EntityDeathEvent firing twice.
```

Comment:
You cannot just skip killing/discarding the entity here if you don't add this logic to the prior called method.
This PR right now prevents the armor stand from dying from anything that isn't void damage or creative attacks.